sled-agent/src/sim/storage.rs

```diff
@@ -1229,6 +1229,9 @@ impl Zpool {
 /// Represents a nested dataset
 pub struct NestedDatasetStorage {
     config: NestedDatasetConfig,
+    // in-memory property for whether this dataset is mounted; typically `true`,
+    // but can be explicitly set to false for some tests
```

Comment:
I know this is a nitpick, but can you humor me?

```suggestion
    // in-memory property for whether this dataset pretends to be mounted; typically `true`,
    // but can be explicitly set to false for some tests
```

Just to clarify: This boolean only changes how this gets reported; it is not used in any real logic beyond "it's a sticky bool of whatever last value you set".